mmv1/templates/terraform/pre_create/dialogflow_set_location.go.erb

```diff
@@ -0,0 +1,13 @@
+
+// extract location from the parent
+stringParts := strings.Split(d.Get("parent").(string), "/")
+if stringParts[2] != "locations" {
+    return fmt.Errorf(
+        "Saw %s when the parent is expected to contains location %s",
+        d.Get("parent"),
+        "projects/{{project}}/locations/{{location}}/...",
+    )
+}
+
+url = strings.Replace(url,"-dialogflow",fmt.Sprintf("%s-dialogflow",stringParts[3]),1)
```

Comment:
How does this replace `{{location}}` in the URL? Does location get filled in with an empty string, just leaving the `-dialogflow`? 

Would it make sense to include `location` as a field on these sub-resources so that we don't have to rely on extracting the location from the parent field?

Reply:
yeah because in the url we specified **{{location}}-dialogflow....**.  when the replaceVars is executed the result will be **-dialogflow** because the location is not part of the resource and the current implementation of replaceVars is to replace only the properties in the resource itself.

From a point of view of User experience I think so is not nice if you have to add the location in every resource, but for sure this simplify the technical implementation. What do you think? In other PR someone recommended  not duplicate the location in every resource